### PR TITLE
29

### PR DESCRIPTION
**Pull request by sanjaymhj**
- added dosage field in Vaccination in coreData
- add the dosage field while adding custom vaccination
- change the vaccination date of the vaccine and reset the notification time
@anmol-lf @bishalg @saugat135 @laxmankhanal @kusalshrestha @vsujan

### CODE REVIEW

File: Vaccination/CoreData/Vaccination+CoreDataProperties.swift [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
+//  Vaccination+CoreDataProperties.swift
+//
+//
+//  Created by Sanjay Maharjan on 4/14/16.
+//
+//
+//  Choose "Create NSManagedObject Subclass…" from the Core Data editor menu
+//  to delete and recreate this implementation file for your updated model.
+//
+
```
Comment (anmol-lf) [inline Vaccination/CoreData/Vaccination+CoreDataProperties.swift]: remove headers

Comment (sanjaymhj) [inline Vaccination/CoreData/Vaccination+CoreDataProperties.swift]: :ok:

File: Vaccination/ViewController/CustomVaccinationVC.swift [hunk 2]
```diff
@@ -79,9 +90,17 @@ class CustomVaccinationVC: UIViewController {
...
 
     func validateRelativeTime(time: String) -> Bool {
+       return isNumber(time)
+    }
+
+    func validateDosage(dosage: String) -> Bool {
+       return isNumber(dosage)
+    }
+
+    func isNumber(input: String) -> Bool {
```
Comment (anmol-lf) [inline Vaccination/ViewController/CustomVaccinationVC.swift]: refactor it something more readable like
func isNumber(input: String) -> Bool {
    guard let number = Double(input) else { return false }
    return !number.isNaN
}

Comment (sanjaymhj) [inline Vaccination/ViewController/CustomVaccinationVC.swift]: :ok:

File: Vaccination/ViewController/MainVC.swift [hunk 3]
```diff
@@ -407,16 +418,28 @@ extension MainVC: UICollectionViewDelegate {
...
+
+    func showDatePickerWithButtons(cell: VaccinationCell) {
+        self.hiddenTextField.inputAccessoryView = KeyBoardToolBar.keyBoardActionButtons(withViewController: self, withActionOnDone: KeyBoardToolBar.keyboardActionSelector(keyBoardRequestor).0, withActionOnCancel: KeyBoardToolBar.keyboardActionSelector(keyBoardRequestor).1)
         self.hiddenTextField.becomeFirstResponder()
         datePickerView.datePickerMode = UIDatePickerMode.Date
         datePickerView.addTarget(self, action: #selector(MainVC.childVaccinatedDate(_:)), forControlEvents: UIControlEvents.ValueChanged)
-        checkedVaccineIndexPath = self.vaccinationListTVC.indexPathForCell(cell)!
-        self.hiddenTextField.inputAccessoryView  = KeyBoardToolBar.keyBoardActionButtons(withViewController: self, withActionOnDone: KeyBoardToolBar.keyboardActionSelector(keyBoardRequestor).0, withActionOnCancel: KeyBoardToolBar.keyboardActionSelector(keyBoardRequestor).1)
+        self.checkedVaccineIndexPath = self.vaccinationListTVC.indexPathForCell(cell)!
+
```
Comment (laxmankhanal) [inline Vaccination/ViewController/MainVC.swift]: remove this extra line.

Comment (sanjaymhj) [inline Vaccination/ViewController/MainVC.swift]: :ok:

**Pull request merged**